python/fledge/services/core/api/filters.py

```diff
@@ -424,8 +425,9 @@ async def delete_filter(request: web.Request) -> web.Response:
         payload = PayloadBuilder().WHERE(['name', '=', filter_name]).payload()
         result = await storage.query_tbl_with_payload("filter_users", payload)
         if len(result["rows"]) != 0:
-            raise TypeError("Filter '{}' found in pipelines".format(filter_name))
-
+            msg = ("The filter '{}' is currently in use within a pipeline. You cannot delete it while it's being used. "
+                   "To proceed, remove it from the pipeline first by updating the pipeline.").format(filter_name)
+            raise ConflictException(msg)
```

Comment:
Cannot we simply raise web.HTTPConflict here with reason and body

Reply:
No, if we need the exact HTTP status code, we should use a return statement instead of raise.